test/async.jl

```diff
@@ -20,7 +20,7 @@
             @test (@mock f()) == "mocked"
 
             notify(c)
-            @test_broken take!(ch) == "original"
+            @test take!(ch) == "original"
```

Comment:
Will need to be condition based upon the Julia version

Reply:
I just disabled the test entirely on old versions.
Does do you want to instead run the test and have it report broken?
I don't see much point if we are not going to fix it on old version ever. (which I guess we could do with some work but its very nontrivial thus why it wasn't done)
Instead maybe we should document that this doesn't work right with async on old julia versions

Reply:
I don't think we need to get this working on older versions. Maybe I'm missing something but where did you disable this test on old versions?

Reply:
I was clearly very out of it when i made this PR. I did not make the change I thought i did